[BUG] Test failure in "SimpleNetty4TransportTests.testHandshakeUpdatesVersion"

**Describe the bug**
Comes from https://github.com/opensearch-project/OpenSearch/pull/902

The test passes at most time, but fails with the specific seed. Failure is only reproducible  in the current `main` branch, and not in `1.x` or `1.0` branch.
```
  2> REPRODUCE WITH: ./gradlew ':modules:transport-netty4:test' --tests "org.opensearch.transport.netty4.SimpleNetty4TransportTests.testHandshakeUpdatesVersion" -Dtests.seed=59DADD0DFC828D39 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1" -Dtests.locale=en-AU -Dtests.timezone=Etc/GMT-14 -Druntime.java=15
  2> java.lang.AssertionError: expected:<7.10.2> but was:<1.1.0>
        at __randomizedtesting.SeedInfo.seed([59DADD0DFC828D39:E8996B23DD77BEC4]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:118)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at org.opensearch.transport.AbstractSimpleTransportTestCase.testHandshakeUpdatesVersion(AbstractSimpleTransportTestCase.java:2016)
...
```
```
Tests with failures:
 - org.opensearch.transport.netty4.SimpleNetty4TransportTests.testHandshakeUpdatesVersion
```

Another assertion error with a different seed (appeared in PR https://github.com/opensearch-project/OpenSearch/pull/913):
```
> Task :modules:transport-netty4:test

REPRODUCE WITH: ./gradlew ':modules:transport-netty4:test' --tests "org.opensearch.transport.netty4.SimpleNetty4TransportTests.testHandshakeUpdatesVersion" -Dtests.seed=425B0821A98A1BA3 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1" -Dtests.locale=es-VE -Dtests.timezone=Asia/Yangon -Druntime.java=15

org.opensearch.transport.netty4.SimpleNetty4TransportTests > testHandshakeUpdatesVersion FAILED
    java.lang.AssertionError: expected:<7.10.2> but was:<1.0.0>
        at __randomizedtesting.SeedInfo.seed([425B0821A98A1BA3:F318BE0F887F285E]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:118)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at org.opensearch.transport.AbstractSimpleTransportTestCase.testHandshakeUpdatesVersion(AbstractSimpleTransportTestCase.java:2016)
...
```

**To Reproduce**
Steps to reproduce the behavior:
1. Clone OpenSearch repository, and checkout `main` branch.
2. Run the command in the root directory:
```
./gradlew ':modules:transport-netty4:test' --tests "org.opensearch.transport.netty4.SimpleNetty4TransportTests.testHandshakeUpdatesVersion" -Dtests.seed=59DADD0DFC828D39
```
or
```
./gradlew ':modules:transport-netty4:test' --tests "org.opensearch.transport.netty4.SimpleNetty4TransportTests.testHandshakeUpdatesVersion" -Dtests.seed=425B0821A98A1BA3
```
3. See the error.

**Expected behavior**
Test passes.

**Plugins**
No plugins installed.

**Screenshots**
(none)

**Host/Environment (please complete the following information):**
 - OS: macOS and Ubuntu

**Additional context**
(none)

--- a/rest-api-spec/src/main/resources/rest-api-spec/test/cat.health/10_basic.yml
+++ b/rest-api-spec/src/main/resources/rest-api-spec/test/cat.health/10_basic.yml
@@ -1,5 +1,8 @@
 ---
 "Help":
+  - skip:
+        version: " - 7.9.99"
+        reason: "discovered_master added in OpenSearch 1.0.0"
   - do:
       cat.health:
         help: true
@@ -27,7 +30,9 @@
 
 ---
 "Empty cluster":
-
+  - skip:
+      version: " - 7.9.99"
+      reason: "discovered_master added in OpenSearch 1.0.0"
   - do:
       cat.health: {}
 
@@ -56,7 +61,9 @@
 
 ---
 "With ts parameter":
-
+  - skip:
+      version: " - 7.9.99"
+      reason: "discovered_master added in OpenSearch 1.0.0"
   - do:
       cat.health:
         ts: false

--- a/rest-api-spec/src/main/resources/rest-api-spec/test/cat.health/10_basic.yml
+++ b/rest-api-spec/src/main/resources/rest-api-spec/test/cat.health/10_basic.yml
@@ -1,7 +1,7 @@
 ---
 "Help":
   - skip:
-        version: " - 7.9.99"
+        version: " - 7.10.99"
         reason: "discovered_master added in OpenSearch 1.0.0"
   - do:
       cat.health:
@@ -31,7 +31,7 @@
 ---
 "Empty cluster":
   - skip:
-      version: " - 7.9.99"
+      version: " - 7.10.99"
       reason: "discovered_master added in OpenSearch 1.0.0"
   - do:
       cat.health: {}
@@ -62,7 +62,7 @@
 ---
 "With ts parameter":
   - skip:
-      version: " - 7.9.99"
+      version: " - 7.10.99"
       reason: "discovered_master added in OpenSearch 1.0.0"
   - do:
       cat.health:

--- a/test/framework/src/main/java/org/opensearch/transport/AbstractSimpleTransportTestCase.java
+++ b/test/framework/src/main/java/org/opensearch/transport/AbstractSimpleTransportTestCase.java
@@ -2013,7 +2013,8 @@ public abstract class AbstractSimpleTransportTestCase extends OpenSearchTestCase
                 TransportRequestOptions.Type.REG,
                 TransportRequestOptions.Type.STATE);
             try (Transport.Connection connection = serviceA.openConnection(node, builder.build())) {
-                assertEquals(connection.getVersion(), version);
+                // OpenSearch 1.0+ in bwc mode should only "upgrade" to Legacy v7.10.2
+                assertEquals(connection.getVersion(), version.onOrAfter(Version.V_1_0_0) ? LegacyESVersion.V_7_10_2 : version);
             }
         }
     }

